### PR TITLE
refactor(common): Use Nullish coalescing assignment when possible.

### PR DESCRIPTION
**Pull request by JeanMeche**
Inspired by #53923, lets simplify code when possible.

### CODE REVIEW

File: packages/common/src/location/hash_location_strategy.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export class HashLocationStrategy extends LocationStrategy implements OnDestroy
     // the hash value is always prefixed with a `#`
     // and if it is empty then it will stay empty
     let path = this._platformLocation.hash;
-    if (path == null) path = '#';
+    path ??= '#';
```
Comment (garrettld) [inline packages/common/src/location/hash_location_strategy.ts:69]: This and the line above could just be `const path = this._platformLocation.hash ?? '#';`

Comment (JeanMeche) [reply packages/common/src/location/hash_location_strategy.ts:69]: Good catch !

File: packages/common/upgrade/test/upgrade_location_test_module.ts [hunk 2]
```diff
@@ -51,9 +51,8 @@ export class LocationUpgradeTestModule {
           useFactory: (appBaseHref?: string) => {
             if (config && config.appBaseHref != null) {
```
Comment (alan-agius4) [inline packages/common/upgrade/test/upgrade_location_test_module.ts:52]:
```suggestion
            if (config?.appBaseHref != null) {
```

File: packages/common/upgrade/test/upgrade_location_test_module.ts [hunk 3]
```diff
@@ -51,9 +51,8 @@ export class LocationUpgradeTestModule {
...
               appBaseHref = config.appBaseHref;
-            } else if (appBaseHref == null) {
-              appBaseHref = '';
             }
+            appBaseHref ??= '';
             return new MockPlatformLocation(
                 {startUrl: config && config.startUrl, appBaseHref: appBaseHref});
```
Comment (alan-agius4) [inline packages/common/upgrade/test/upgrade_location_test_module.ts:57]:
```suggestion
                {startUrl: config?.startUrl, appBaseHref: appBaseHref});
```

File: packages/common/src/location/hash_location_strategy.ts [hunk 4]
```diff
@@ -65,8 +65,7 @@ export class HashLocationStrategy extends LocationStrategy implements OnDestroy
   override path(includeHash: boolean = false): string {
     // the hash value is always prefixed with a `#`
     // and if it is empty then it will stay empty
-    let path = this._platformLocation.hash;
-    if (path == null) path = '#';
+    let path = this._platformLocation.hash ?? '#';
```
Comment (dgp1130) [inline packages/common/src/location/hash_location_strategy.ts:68]: Nit: This can be `const` now.

**Review by dgp1130 — APPROVED** [review]
Reviewed for: fw-i18n

**Review by dylhunn — APPROVED** [review]

**Review by jessicajaniuk — APPROVED** [review]

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit 89ab1c18a19c37e82652af609dd4977d0535fa98.

**Pull request closed without merging**